File: study_ws/src/cpp_pkg/src/NumberPublisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int64.hpp" 
#include <cstdlib>  // For rand()


class NumberPublisher : public rclcpp::Node {
public:
    NumberPublisher() : Node("number_publisher") {
        publisher_ = this->create_publisher<std_msgs::msg::Int64>("number", 10);
        while true{
            try{
                this->declare_parameter<int>("publishFrequency", 1);
                float period_ = (1000/this->get_parameter("publishFrequency").get_value<int>());
                break;
            } catch (const std::exception &e){
                RCLCPP_WARN(this->get_logger(), "Exception :  %s", e.what());
                //sleep for 2 sec
            }
        }

        timer_ = this->create_wall_timer(
            std::chrono::milliseconds(period_), std::bind(&NumberPublisher::publishNumber, this));
    }

private:
    void publishNumber() {
        auto message = std_msgs::msg::Int64();
        message.data = rand() % 100;  // Generate a random number between 0 and 99
        RCLCPP_INFO(this->get_logger(), "Publishing: '%ld'", message.data);
        publisher_->publish(message);
    }
    float period_;
    rclcpp::Publisher<std_msgs::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char *argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<NumberPublisher>());
    rclcpp::shutdown();
    return 0;
}

File: study_ws/src/py_pkg/py_pkg/numberPublisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import Int64

# Python Specifics
import random
from functools import partial

NodeName = "PublisherNode"

class numPub(Node): 
    def __init__(self):
        super().__init__(NodeName)
        self.get_logger().info('NumberPublisher Has been initiated')
        # Tried ros2Parameters in this region..
        self.declare_parameter('number_to_publish', 10)
        self.declare_parameter('publishFrequency', 1)
        self.period = 1/self.get_parameter('publishFrequency').value
        #*********************************************
        self.number_publisher_ = self.create_publisher(Int64, 'number', 10)
        self.timer_callback = self.create_timer(self.period, self.publishNumber)
    
    def publishNumber(self):
        msg = Int64()
        #random.randint(1, 100)
        try:
            msg.data = self.get_parameter('number_to_publish').value
            self.get_logger().info('Publishing number :' + str(msg.data))
            self.number_publisher_.publish(msg)            
        except Exception as e:
            self.get_logger().error(f'Ran into an error : {e}')

def main(args=None):
    rclpy.init(args=args)
    node = numPub()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.get_logger().info('Detected KeyboardInterrupt. Exiting...')
    finally:
        # Clean up the node and shutdown rclpy
        if rclpy.ok():
            node.destroy_node()
            rclpy.shutdown()


if __name__ == "__main__":
    main()
